app/controllers/users/sessions_controller.rb

```diff
@@ -13,19 +16,26 @@ def create
   end
 
   def sign_in_with_token
-    user = User.find_by(login_token: params[:login_token])
-
-    if user.present? && Time.zone.now < user.login_token_valid_until
-      user.update!(login_token: nil, login_token_valid_until: 1.year.ago)
-      sign_in(user, scope: :user)
-      redirect_to helpers.profile_dashboard_url(user)
-    else
-      flash[:alert] = "There was an error while logging you in. Please enter your email again."
-      redirect_to new_user_session_path
-    end
+    @user.update!(login_token: nil, login_token_valid_until: 1.year.ago)
+    sign_in(@user, scope: :user)
+    redirect_to_dashboard
   end
 
   def redirect_from_magic_link
     @login_token = params[:login_token] if params[:login_token].present?
   end
+
+private
+
+  def redirect_to_dashboard
+    redirect_to helpers.profile_dashboard_url(current_user) if current_user.present?
```

Comment:
Is `current_user` always gonna be the same as `@user`? I think so, so maybe let's just use `@user` here?

Reply:
Not in the before actions 🙂